File: Java/029.java
import java.math.BigInteger;
import java.util.HashSet;
import java.util.Set;

public class Problem029 {

    public static int generateTerms(int a, int b) {
        Set<BigInteger> terms = new HashSet<>();
        for (int i = 2; i <= a; i++) {
            for (int j = 2; j <= b; j++) {
                terms.add(BigInteger.valueOf(i).pow(j));
            }
        }
        return terms.size();
    }

    public static void main(String[] args) {
        int res = generateTerms(100, 100);
        System.out.println(res);
    }
}

File: Rust/029.rs
use num_bigint::BigUint;
use std::collections::HashSet;

fn generate_terms(a_max: u32, b_max: u32) -> usize {
    let mut terms = HashSet::new();
    for a in 2..=a_max {
        for b in 2..=b_max {
            terms.insert(BigUint::from(a).pow(b));
        }
    }
    terms.len()
}

fn main() {
    let res = generate_terms(100, 100);
    println!("{}", res);
}
